### PR TITLE
Add test for pgenerate

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Comment by kshyatt** [comment]
Is the AV failure my fault? Doesn't look like it to me...

File: test/parallel_exec.jl [hunk 1]
```diff
@@ -774,6 +774,17 @@ end
 # Test pmap with a generator type iterator
 @test [1:100...] == pmap(x->x, Base.Generator(x->(sleep(0.0001); x), 1:100))
 
+# Test pgenerate
+a = rand(4,4)
+b = fill(a, 4)
+c = collect(Base.pgenerate(svdfact, b))
```
Comment (amitmurthy) [inline test/parallel_exec.jl]: As `pgenerate` is expected to process elements a) asynchronously and b) return them in the correct order, I would improve this test by incorporating a random delay:

`c = collect(Base.pgenerate(x->(sleep(rand()*0.1);svdfact(x);, b))`

Comment (kshyatt) [inline test/parallel_exec.jl]: oooooh nice! will do :)

I can change the input arg to have different elements to make sure the return order is correct as well

Comment (amitmurthy) [inline test/parallel_exec.jl]: Yes, that is a good idea.

**Comment by kshyatt** [comment]
@amitmurthy good to go?

File: test/parallel_exec.jl [hunk 2]
```diff
@@ -774,6 +774,19 @@ end
...
+a3 = rand(4,4)
+b = [a1, a2, a3]
+c = collect(Base.pgenerate(x->(sleep(rand()*0.1); svdfact(x)), b))
+svda = [svdfact(a1), svdfact(a2), svdfact(a3)]
+for i in 1:3
+    @test c[i][:U] ≈ svda[i][:U]
+    @test c[i][:S] ≈ svda[i][:S]
+    @test c[i][:V] ≈ svda[i][:V]
+end
+
```
Comment (amitmurthy) [inline test/parallel_exec.jl]: Nitpicking. But can we increase the collection length? Something like
```
n=10
a=[rand(4,4) for _ in 1:n]
....
```

**Comment by amitmurthy** [comment]
Looks good.

**Pull request merged**